### DISCUSSION TITLE
How to solve Mock context on EMR

### DISCUSSION THREAD

**Discussion by HPicatto** [dagster-io/dagster#19278] [Q&A]
Hi, I've a pipeline with assets that can be executed either locally or on EMR. I know the code works locally. but when the task is running on EMR it cannot find the context correctly. Istead of actually using the emr context it does some kind of mock:

> :: loading settings :: url = jar:file:/usr/lib/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
> *** Engine: Engine.EMR ***
> Using EMR context
> /usr/local/lib/python3.9/site-packages/dagster_pipes/__init__.py:398: DagsterPipesWarning: This process was not launched by a Dagster orchestration process. All calls to the `dagster-pipes` context or attempts to initialize `dagster-pipes` abstractions are no-ops.
>   warnings.warn(
> Traceback (most recent call last):
>   File "/mnt/tmp/spark-0a7bc0eb-d0a0-41d2-a8f2-50d51ab029f3/01_nodes_only.py", line 168, in <module>
>     NodesOnly()
>   File "/usr/local/lib/python3.9/site-packages/ascii_library/orchestration/pipes/spark_script_abc.py", line 87, in __init__
>     engine = Engine(context.get_extra("engine"))
>   File "/usr/lib64/python3.9/enum.py", line 384, in __call__
>     return cls.__new__(cls, value)
>   File "/usr/lib64/python3.9/enum.py", line 702, in __new__
>     raise ve_exc
> ValueError: <MagicMock name='mock.get_extra()' id='140405404586480'> is not a valid Engine 

any idea why it mocks instead of retrieving the right context when I do context = PipesContext.get()

**Comment by jamiedemaria** [comment]
Hey @HPicatto can you share the asset you're running that results in this error?
